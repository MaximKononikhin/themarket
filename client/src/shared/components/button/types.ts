import { ButtonHTMLAttributes, ReactNode } from "react";

export interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
    /** Контент кнопки **/
    children?: ReactNode;
    /** id **/
    testId?: string;
    className?: string;
    wide?: boolean;
}
