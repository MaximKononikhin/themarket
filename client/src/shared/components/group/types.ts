export type GroupProps = {
    gap?: number;
    direction: "column" | "row";
    alignItems?: "center" | "flex-start" | "flex-end" | "end";
    width?: string;
    className?: string;
    margin?: string;
    testId?: string;
};
