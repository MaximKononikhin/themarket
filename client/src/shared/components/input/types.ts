import { InputHTMLAttributes, ReactNode } from "react";

import { Type } from "../typography/type";

export interface InputProps
    extends InputHTMLAttributes<HTMLInputElement | HTMLTextAreaElement> {
    /** Тип инпута **/
    design?: "input" | "textarea";
    /** Ширина инпута **/
    width?: "wide" | "normal" | "small";
    /** Лейбл **/
    label?: string | ReactNode;
    /** Плейсхолдер **/
    placeholder?: string;
    /** Текст ошибки **/
    error?: string;
    /** Тип ошибки **/
    errorType?: "absolute" | "static";
    /** labelStyle **/
    labelStyle?: Extract<Type, "header-1" | "header-2">;
    /** стилизация **/
    className?: string;
    /** тест-айди **/
    testId?: string;
}
