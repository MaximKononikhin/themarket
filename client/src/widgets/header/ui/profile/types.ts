import { UserEntity } from "@shared/api";

export type ProfileProps = {
    user: UserEntity;
};
