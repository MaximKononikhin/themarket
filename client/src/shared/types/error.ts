export type IError = {
    statusCode?: number;
    message: string;
};
