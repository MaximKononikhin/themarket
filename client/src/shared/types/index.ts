export * from "./utility";
export * from "./error";
