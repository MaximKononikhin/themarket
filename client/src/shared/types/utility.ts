export type Maybe<T, S = null> = T | S;
