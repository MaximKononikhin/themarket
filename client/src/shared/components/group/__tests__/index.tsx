import React from "react";

import { render } from "@testing-library/react";

import { Group } from "../index";
import { GroupProps } from "../types";

const GROUP_PROPS: GroupProps = {
    gap: 10,
    direction: "column",
    alignItems: "center",
    width: "100px",
    margin: "0px",
};

const TEST_ID = "component";

describe("Group component", () => {
    it("should render component and has proper styles", () => {
        const { getByTestId } = render(
            <Group {...GROUP_PROPS} testId={TEST_ID}>
                <div />
            </Group>
        );

        const element = getByTestId(TEST_ID);

        expect(element).toBeInTheDocument();

        Object.entries(GROUP_PROPS).forEach(([key, value]) => {
            if (key === "alignItems") {
                expect(element).toHaveStyle({ "align-items": value });
                return;
            }
            if (key === "gap") {
                expect(element).toHaveStyle({ [key]: `${value}px` });
                return;
            }
            if (key === "direction") {
                expect(element).toHaveStyle({ "flex-direction": value });
                return;
            }

            expect(element).toHaveStyle({ [key]: value });
        });

        expect(element).toMatchSnapshot();
    });
});
