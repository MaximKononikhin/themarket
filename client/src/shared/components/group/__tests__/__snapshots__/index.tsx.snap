// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Group component should render component and has proper styles 1`] = `
<div
  class="group"
  data-testid="component"
  style="gap: 10px; align-items: center; flex-direction: column; width: 100px; margin: 0px;"
>
  <div />
</div>
`;
